File: app/src/main/java/org/wikipedia/dataclient/mwapi/MwPostResponse.java
package org.wikipedia.dataclient.mwapi;

import androidx.annotation.Nullable;

import org.wikipedia.dataclient.wikidata.Entities;

@SuppressWarnings("unused")
public class MwPostResponse extends MwResponse {
    @Nullable private MwQueryPage pageinfo;
    @Nullable private Entities.Entity entity;
    @Nullable private String options;
    private int success;

    public boolean success(@Nullable String result) {
        return "success".equals(result);
    }

    @Nullable public String getOptions() {
        return options;
    }

    public int getSuccessVal() {
        return success;
    }

    @Nullable public MwQueryPage getPageInfo() {
        return pageinfo;
    }

    @Nullable public Entities.Entity getEntity() {
        return entity;
    }
}

File: app/src/main/java/org/wikipedia/dataclient/mwapi/MwPostResponse.kt
package org.wikipedia.dataclient.mwapi

import kotlinx.serialization.Serializable
import org.wikipedia.auth.AccountUtil

@Serializable
open class MwPostResponse : MwResponse() {
    val pageInfo: MwQueryPage? = null
    val options: String? = null
    val success = 0

    init {
        // If the user gets assigned a temporary account as a result of a POST action, then
        // record the day of the account creation.
        AccountUtil.maybeSetTempAccountDay()
    }
}
